core/src/main/java/com/linecorp/armeria/server/AbstractRouteBuilder.java

```diff
@@ -0,0 +1,509 @@
+    }
+
+    /**
+     * Sets the {@link ContentPreviewerFactory} for creating a {@link ContentPreviewer} which produces the
+     * preview with the maximum {@code length} limit for an HTTP request/response of the
+     * {@linkplain #service(Service) service}. The previewer is enabled only if the content type of an HTTP
+     * request/response meets any of the following conditions:
+     * <ul>
+     *     <li>when it matches {@code text/*} or {@code application/x-www-form-urlencoded}</li>
+     *     <li>when its charset has been specified</li>
+     *     <li>when its subtype is {@code "xml"} or {@code "json"}</li>
+     *     <li>when its subtype ends with {@code "+xml"} or {@code "+json"}</li>
+     * </ul>
+     * @param length the maximum length of the preview
+     * @param defaultCharset the default charset used when a charset is not specified in the
+     *                       {@code "content-type"} header
+     */
+    public AbstractRouteBuilder<B> contentPreview(int length, Charset defaultCharset) {
+        return contentPreviewerFactory(ContentPreviewerFactory.ofText(length, defaultCharset));
+    }
+
+    /**
+     * Sets the {@link ContentPreviewerFactory} for an HTTP request/response of the
+     * {@linkplain #service(Service) service}.
+     */
+    public AbstractRouteBuilder<B> contentPreviewerFactory(ContentPreviewerFactory factory) {
+        requestContentPreviewerFactory(factory);
+        responseContentPreviewerFactory(factory);
+        return this;
+    }
+
+    /**
+     * Decorates a {@link Service} with the specified {@code decorator}.
+     *
+     * @param decorator the {@link Function} that decorates the {@link Service}
+     * @param <T> the type of the {@link Service} being decorated
+     * @param <R> the type of the {@link Service} {@code decorator} will produce
+     */
+    public <T extends Service<HttpRequest, HttpResponse>, R extends Service<HttpRequest, HttpResponse>>
+    AbstractRouteBuilder<B> decorator(Function<T, R> decorator) {
+        requireNonNull(decorator, "decorator");
+
+        @SuppressWarnings("unchecked")
+        final Function<Service<HttpRequest, HttpResponse>, Service<HttpRequest, HttpResponse>> castDecorator =
+                (Function<Service<HttpRequest, HttpResponse>, Service<HttpRequest, HttpResponse>>) decorator;
+
+        if (this.decorator != null) {
+            this.decorator = this.decorator.andThen(castDecorator);
+        } else {
+            this.decorator = castDecorator;
+        }
+
+        return this;
+    }
+
+    /**
+     * Sets the {@link Service} and returns the {@link B} that this builder was created from.
+     *
+     * @throws IllegalStateException if the path that the {@link Service} will be bound to is not specified
+     *
+     * @see #path(String)
+     * @see #pathUnder(String)
+     * @see #pathMapping(PathMapping)
+     */
+    public abstract B service(Service<HttpRequest, HttpResponse> service);
```

Comment:
I guess you can just return `Object` and remove the type parameter `B` completely.